### DISCUSSION TITLE
How can I access WIN system variables in WSL?

### DISCUSSION THREAD

**Discussion by FrankBergemann** [microsoft/WSL#4927] [General]
I have question on WSL bash shell usage on WIN10: 
How can I access WIN system variables e.g. JAVA_HOME, MVN_HOME etc. in a WSL bash shell?
Is there some way to import all of them?
I even have trouble handling the blank between "Program" and "Files" in C:\Program Files\Java\jdk-11.0.4 when trying to set in JAVA_HOME in .profile manually.
Can anyone help?

**Comment by therealkenc** [comment]
[WSLENV](https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/). You can also get them with:

```
$ JAVA_HOME=$(pushd /mnt/c > /dev/null; cmd.exe /c echo %JAVA_HOME%; popd > /dev/null)
$ echo "$JAVA_HOME"
```

![image](https://user-images.githubusercontent.com/7049700/75184716-af881d00-56f9-11ea-9610-df4f8b1b5a18.png)

As for undefined "handling the blank", you'll get better eyeballs on shell programming in [stackoverflow](https://stackoverflow.com/) and the like.
